assayist/processor/base.py

```diff
@@ -1,16 +1,52 @@
 # SPDX-License-Identifier: GPL-3.0+
 
 from abc import ABC, abstractmethod
+from functools import cmp_to_key
 import json
 import os
 
-import neomodel
+from neomodel import db, ZeroOrOne, One
+from pkg_resources import parse_version
+import rpm
 
 from assayist.common.models import content, source
 from assayist.processor.configuration import config
 from assayist.processor.logging import log
 
 
+def rpm_compare(x, y):
+    """
+    Compare two rpm canonical_versions.
+
+    :param str x: The first rpm version
+    :param str y: The second rpm version
```

Comment:
I guess the expected type is `SourceLocation` since you're expecting `x` and `y` to have a `canonical_version` attribute.